### DISCUSSION TITLE
Using multiple envrionment.yml files

### DISCUSSION THREAD

**Discussion by stefanbringuier** [rodluger/showyourwork#413] [Q&A] [answered]
I'm trying to create several rules where each uses a different conda `environment.yml` file. I need to do so because I'm using several `pip` packages that all have very stringent PyTorch version requirements. The issue I'm running into is that even if I specify a different `.yml` file name for the `conda` flag, it seems to always default to `environment.yml`. For example, the following does not work for me:

```snakemake
rule simulation_1:
    input:
        "src/scripts/simulation_1.py"
    output:
        "src/data/simulation_1.data"
    conda:
        "environment_sim1.yml"
    script:
        "src/scripts/simulation_1.py

rule simulation_2:
    input:
        "src/scripts/simulation_2.py"
    output:
        "src/data/simulation_2.data"
    conda:
        "environment_sim2.yml"
    script:
        "src/scripts/simulation_2.py

rule plot:
    input:
       "src/data/simulation_1.data"
       "src/data/simulation_2.data"
       "src/scripts/plot.py"
    output:
       "src/data/results.png"
    conda:
        "environment_plots.yml"
    script:
        "scr/scripts/plot.py
```

I don't get a specific error/issue from the defined `conda` env in the `Snakemake` file, but when I run `showyourwork build`, the scripts such as `simulation_1.py` fail to import the required Python modules that are specified in the `environment_sim1.yml` file. Thus, it seems to be ignoring the environment file. Furthermore, I don't see a step in the `showyourwork build` output indicating it is setting up the proper environment.

Is this ultimately a limitation of `SnakeMake`?

**Comment by dfm** [comment]
I would expect this to absolutely work. Would you be able to point me to the project or create a minimal project that demonstrates the problem?

**Reply by stefanbringuier** [reply, marked as answer]
Okay, so I think I found the issue.

Indeed having unique `conda` environment `yml` files does work. However, it appears `showyourwork` requires, at minimum, the default `environment.yml` file in the main project folder. This was my issue, I had removed it.

**Reply by dfm** [reply]
Thanks for the update - glad you got it working!
